src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,74 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { useState } from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type TodoFilterProps = {
+  onFilter: (status: string) => void,
+  onTitleFilter: (title: string) => void,
+  onClearFilter: () => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter
+  = ({ onFilter, onTitleFilter, onClearFilter } : TodoFilterProps) => {
+    const [query, setQuery] = useState<string>('');
+    const [filter, setFilter] = useState<string>('all');
+    const handleSelect: React.ChangeEventHandler<HTMLSelectElement>
+    = (event) => {
+      setFilter(event.target.value);
+      onFilter(event.target.value);
```

Comment:
who do we need separate states for that?

Reply:
implement previous suggestions.